phoenix-core/src/it/java/org/apache/phoenix/end2end/AlterAddCascadeIndexIT.java

```diff
@@ -0,0 +1,272 @@
+        StringBuilder optionBuilder = new StringBuilder();
+        if (!columnEncoded) {
+            optionBuilder.append("COLUMN_ENCODED_BYTES=0");
+        }
+        if (!mutable && !columnEncoded) {
+            optionBuilder.append(",IMMUTABLE_ROWS=true");
+        } else if (!mutable && columnEncoded) {
+            optionBuilder.append("IMMUTABLE_ROWS=true");
+        }
+        if (!columnEncoded && !mutable) {
+            optionBuilder.append(",IMMUTABLE_STORAGE_SCHEME="+ PTableImpl.ImmutableStorageScheme.ONE_CELL_PER_COLUMN);
+        }
+        this.tableDDLOptions = optionBuilder.toString();
+    }
+
+    @Parameters(name="AlterAddCascadeIndexIT_isViewIndex={0},isLocalIndex={1},mutable={2},columnEncoded={3}")
+    public static Collection<Object[]> data() {
+        return Arrays.asList(new Object[][] {
+            { true, true, true, true},
+            { true, true, true, false},
+            { true, true, false, true},
+            { true, true, false, false},
+            { true, false, true, true},
+            { true, false, true, false},
+            { true, false, false, true},
+            { true, false, false, false},
+            { false, true, true, true},
+            { false, true, true, false},
+            { false, true, false, true},
+            { false, true, false, false},
+            { false, false, true, true},
+            { false, false, true, false},
+            { false, false, false, true},
+            { false, false, false, false}
+        });
+    }
+
+    @Before
+    public void setup() throws SQLException {
+        prop = new Properties();
+        conn = DriverManager.getConnection(getUrl(), prop);
+        conn.setAutoCommit(true);
+        conn.createStatement().execute("CREATE TABLE IF NOT EXISTS us_population (\n" +
+                "      state CHAR(2) NOT NULL,\n" +
+                "      city VARCHAR NOT NULL,\n" +
+                "      population BIGINT,\n" +
+                "      CONSTRAINT my_pk PRIMARY KEY (state, city)) " + tableDDLOptions);
+
+        loadData();
+
+        if(isViewIndex) {
+            conn.createStatement().execute("CREATE VIEW IF NOT EXISTS us_population_gv" +
+                    "(city_area INTEGER, avg_fam_size INTEGER) AS " +
+                    "SELECT * FROM us_population WHERE state = 'CA'");
+
+            conn.createStatement().execute("CREATE "+(isLocalIndex ? "LOCAL " : "") +"INDEX IF NOT EXISTS us_population_gv_gi ON " +
+                    "us_population_gv (city_area) INCLUDE (population)");
+            conn.createStatement().execute("CREATE "+(isLocalIndex ? "LOCAL " : "")+"INDEX IF NOT EXISTS us_population_gv_gi_2 ON " +
+                    "us_population_gv (avg_fam_size) INCLUDE (population)");
+            databaseObjectName = "us_population_gv";
+            indexesName = "us_population_gv_gi, us_population_gv_gi_2";
+            planOutput = isLocalIndex? "US_POPULATION" : "_IDX_US_POPULATION";
```

Comment:
This is not exactly plan output. I recommend either using full plan output or just rename the variable to be actualUsedTable